test/pytest/test_packages.py

```diff
@@ -0,0 +1,32 @@
+# This file is part of Cockpit.
+#
+# Copyright (C) 2023 Red Hat, Inc.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+import pytest
+
+from cockpit.packages import parse_accept_language
+
+
+TEST_ACCEPT_LANGUAGES = [
+    ('de-at, zh-CH, en', [('de-at', '1.0'), ('zh-CH', '0.99'), ('en', '0.98')]),
+    ('es-es, nl;q=0.8, fr;q=0.9', [('es-es', '1.0',), ('fr', '0.9'), ('nl', '0.8')]),
+    ('fr-CH, fr;q=0.9, en;q=0.8, de;q=0.7, *;q=0.5', [('fr-CH', '1.0'), ('fr', '0.9'), ('en', '0.8'), ('de', '0.7'), ('*', '0.5')])
+]
+
+
+@pytest.mark.parametrize("test_input,expected", TEST_ACCEPT_LANGUAGES)
```

Comment:
...and I feel like this is probably over-parameterized.  Let's have @martinpitt referee on this one.

Reply:
Not sure what you mean here, but three scenarios feels quite okay, and they do cover a lot of corner cases.